src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -58,6 +60,14 @@ export const setDatasetActionCreator = (
     dispatch(setPromptModeIsAvailable(newPromptModeIsAvailable));
   }
 
+  const newSummaryAgentIsAvailable = await getSummaryAgentIsAvailable(
+    services,
+    query.dataset?.dataSource?.id!
```

Comment:
what happens if dataset or dataSource is undefined?

Reply:
That's the question I would like @kavilla to answer

Reply:
It will check for local cluster and determine if there is a summary agent for it.